Formula/gnuradio.rb

```diff
@@ -1,46 +1,48 @@
 class Gnuradio < Formula
   desc "SDK providing the signal processing runtime and processing blocks"
-  homepage "https://gnuradio.squarespace.com/"
-  url "https://gnuradio.org/releases/gnuradio/gnuradio-3.7.9.1.tar.gz"
-  sha256 "9c06f0f1ec14113203e0486fd526dd46ecef216dfe42f12d78d9b781b1ef967e"
-  revision 2
+  homepage "http://gnuradio.org/"
+  url "https://gnuradio.org/releases/gnuradio/gnuradio-3.7.10.1.tar.gz"
+  sha256 "63d7b65cc4abe22f47b8f41caaf7370a0a502b91e36e29901ba03e8838ab4937"
+  head "https://github.com/gnuradio/gnuradio.git"
 
   bottle do
     sha256 "7f2b54d889d4d568736e9ae8221b0a652015d6abb18b1466a5637f92a007884f" => :sierra
     sha256 "7b70a05aeeb12d4457c81093880786468210d8dc1a409bc1181d6910607e3b3f" => :el_capitan
     sha256 "464555aaac52b55b9e09ec2b0b2e0e4cabc024b655a1bf815d7ddde59aec92fe" => :yosemite
   end
 
+  # Fixes linkage of python (swig) bindings directly to python
+  # https://github.com/gnuradio/gnuradio/pull/1146
+  # Merged into master, probably will be in next release
+  patch do
+    url "https://github.com/gnuradio/gnuradio/pull/1146.patch"
+    sha256 "fbf9842292cc1c2cfcf708d648d01c9a42fb98c5bba542a272dcdac504454d26"
+  end
+
   option :universal
 
   depends_on "pkg-config" => :build
 
-  depends_on :python if MacOS.version <= :snow_leopard
+  depends_on :python => :recommended
   depends_on "boost"
   depends_on "cppunit"
   depends_on "fftw"
   depends_on "gsl"
   depends_on "zeromq"
+  depends_on "numpy" if build.with? :python
+  depends_on "swig" => :build if build.with? :python
+  depends_on "cmake" => :build
 
   # For documentation
-  depends_on "doxygen" => :build
-  depends_on "sphinx-doc" => :build
+  depends_on "doxygen" => [:build, :optional]
+  depends_on "sphinx-doc" => [:build, :optional]
 
   depends_on "uhd" => :recommended
-  depends_on "sdl" => :recommended
-  depends_on "jack" => :recommended
+  depends_on "sdl" => :optional
+  depends_on "jack" => :optional
```

Comment:
Based on [this discussion](https://github.com/Homebrew/homebrew-core/pull/7976#discussion_r92953220)